### PR TITLE
Satisfy clippy

### PR DESCRIPTION
**Pull request by Urhengulas**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -323,7 +316,7 @@ fn notmain() -> Result<i32, anyhow::Error> {
     } else {
         // program lives in Flash
         let size = program_size_of(&elf);
-        log::info!("flashing program ({:.02} KiB)", size as f64 / 1024 as f64);
+        log::info!("flashing program ({:.02} KiB)", size as f64 / 1024_f64);
```
Comment (jonas-schievink) [inline src/main.rs:319]: Even simpler:

```suggestion
        log::info!("flashing program ({:.02} KiB)", size as f64 / 1024.0);
```

File: src/main.rs [hunk 2]
```diff
@@ -680,6 +673,7 @@ impl<'c, 'probe> Registers<'c, 'probe> {
     }
 }
 
+#[allow(clippy::too_many_arguments)]
```
Comment (jonas-schievink) [inline src/main.rs:676]: This is something we should fix at some point, the code in probe-run has become pretty messy

```suggestion
#[allow(clippy::too_many_arguments)] // FIXME: clean this up
```

Comment (Urhengulas) [reply src/main.rs:676]: Is this type of comment automatically recognized by some tool?

Comment (Urhengulas) [reply src/main.rs]: Or should we maybe just open an issue for it, since this is more visible?

**Comment by Urhengulas** [comment]
Thanks @jonas-schievink

**Pull request merged**